File: 169.majorityElement.go
package main

func majorityElement(nums []int) int {
	times := 0
	id := -1

	for i := 0; i < len(nums); i++ {
		num := nums[i]

		if times == 0 {
			id = num
		}

		if id != num {
			times--
		} else {
			times++
		}
	}

	return id
}

File: 169.majorityElement.c
#include "shdstd.h"

int majorityElement(int *nums, int size)
{
	int times = 0;
	int id;
	for (int i = 0; i < size; ++i) {
		int num = nums[i];

		if (times == 0) {
			id = num;
		}

		if (id != num) {
			--times;
		} else {
			++times;
		}
	}

	return id;
}
